plugin-server/tests/worker/ingestion/person-state.test.ts

```diff
@@ -65,606 +68,1719 @@ describe('PersonState.update()', () => {
-                }),
+                    id: expect.any(Number),
+                    uuid: uuid.toString(),
+                    properties: { foo: 'bar' },
+                    created_at: timestamp,
+                    version: 0,
+                    is_identified: false, // is_identified is not tied to $identify calls but used for safeguarding bad merges
+                })
+            )
+
+            // verify Postgres distinct_ids
+            const distinctIds = await hub.db.fetchDistinctIdValues(persons[0])
+            expect(distinctIds).toEqual(expect.arrayContaining(['new-user']))
+
+            // verify ClickHouse persons
+            const clickhousePersons = await delayUntilEventIngested(fetchPersonsRows)
+            expect(clickhousePersons.length).toEqual(1)
+            expect(clickhousePersons[0]).toEqual(
                 expect.objectContaining({
                     id: uuid.toString(),
-                    is_deleted: 1,
-                    version: 100,
-                }),
-            ])
-        )
+                    properties: JSON.stringify({ foo: 'bar' }),
+                    created_at: timestampch,
+                    version: 0,
+                    is_identified: 0,
+                })
+            )
 
-        expect(hub.personManager.isNewPerson).toHaveBeenCalledTimes(0)
-        expect(hub.db.fetchPerson).toHaveBeenCalledTimes(2)
-    })
+            // verify ClickHouse distinct_ids
+            await delayUntilEventIngested(fetchDistinctIdsRows)
+            const clickHouseDistinctIds = await hub.db.fetchDistinctIdValues(persons[0], Database.ClickHouse)
+            expect(clickHouseDistinctIds).toEqual(expect.arrayContaining(['new-user']))
 
-    it('updates person properties when other thread merges the user', async () => {
-        const cachedPerson = await hub.db.createPerson(
-            timestamp,
-            { a: 1, b: 2 },
-            {},
-            {},
-            2,
-            null,
-            false,
-            uuid.toString(),
-            ['old-user']
-        )
-        await hub.db.createPerson(timestamp, {}, {}, {}, 2, null, false, uuid2.toString(), ['new-user'])
-        const mergedPersonContainer = await personState({
-            event: '$identify',
-            distinct_id: 'new-user',
-            properties: {
-                $anon_distinct_id: 'old-user',
-            },
-        }).update()
-        const mergedPerson = await mergedPersonContainer.get()
-        // Prerequisite for the test - UUID changes
-        expect(mergedPerson!.uuid).not.toEqual(cachedPerson.uuid)
-
-        console.log('---')
-        jest.mocked(hub.db.fetchPerson).mockClear() // Reset counter
-
-        const personContainer = await personState(
-            {
-                event: '$pageview',
+            // verify personContainer
+            expect(persons[0]).toEqual(await personContainer.get())
+        })
+
+        it('creates person with anon_distinct_id', async () => {
```

Comment:
Test description needs work
```suggestion
        it('creates person with two distinct_ids when passed a $anon_distinct_id', async () => {
```